cloudinit/ssh_util.py

```diff
@@ -17,27 +17,31 @@
 # See: man sshd_config
 DEF_SSHD_CFG = "/etc/ssh/sshd_config"
 
-# taken from OpenSSH source openssh-7.3p1/sshkey.c:
+# taken from OpenSSH source openssh-8.3p1/sshkey.c:
```

Comment:
Looking at https://github.com/canonical/cloud-init/pull/487/commits/8b95cc8c1ac75aff91a4d6bb70f8c9527a72e9d9: I'm not an OpenSSH expert, but an examination of the code suggests that this and `rsa-sha2-512` are keys, not certificates:

```c
	{ "rsa-sha2-256", "RSA", NULL, KEY_RSA, 0, 0, 1 },
	{ "rsa-sha2-512", "RSA", NULL, KEY_RSA, 0, 0, 1 },
```

(Compare to:

```c
	{ "ssh-rsa-cert-v01@openssh.com", "RSA-CERT", NULL,
	    KEY_RSA_CERT, 0, 1, 0 },
```
)

However, looking at https://superuser.com/a/1488613 and https://github.com/openssh/openssh-portable/blob/master/PROTOCOL.certkeys suggests that they are not actually key types to be stored on disk, but used exclusively during key exchange.

So: I think excluding these types from the list is correct, but I'd like to see this explanatory comment expanded to include this factor. (Put differently: we are no longer simply "taking" the list from `sshkey.c`, we performing some filtering on it before updating this list, so the comment is inaccurate. I'd like us to capture the filtering for the next person who comes along to update this list.)

Reply:
(Thinking about it, we should probably also include a pointer to the list in the docs, so the next person also knows to update that list.)

Reply:
You are correct, they are not stored on the disk. I talked to @djmdjm, whom told me that `webauthn-sk-ecdsa-sha2-nistp256@openssh.com` and the like of `rsa-sha2-256` are not key types, but signature types.

If you take a look at these lines in https://github.com/openssh/openssh-portable/blob/master/sshkey.c#L97-L105
```C
struct keytype {
	const char *name;
	const char *shortname;
	const char *sigalg;
	int type;
	int nid;
	int cert;
	int sigonly;
};
```

the last int signifies signature type, if I'm not 100% wrong. So that's why I removed those in 8b95cc8

I think it's a good idea to leave a comment so that the next person knows to filter out the ones that has the sigonly flag on them